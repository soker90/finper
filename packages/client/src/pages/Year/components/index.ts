export { default as Header } from './Header'
export { default as YearTable } from './YearTable'
export { default as YearCard } from './YearCard'
