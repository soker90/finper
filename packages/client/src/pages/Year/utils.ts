export const isSameDate = (year?: string): boolean => {
    const now = new Date()
    return (year === now.getFullYear().toString())
}

export const getUrlYear = (year?: number): string => {
    return `/presupuestos/${year || new Date().getFullYear()}`
}
