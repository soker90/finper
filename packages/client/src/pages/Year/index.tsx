import { useParams } from 'react-router'

import { useBudgetsYear } from './hooks'

import { Header, YearCard, YearTable } from './components'
import { Grid, Typography } from '@mui/material'
import { MainCard } from 'components'

const Budgets = () => {
  const { year } = useParams()
  const { expenses, incomes } = useBudgetsYear({ year })

  const balance = (incomes?.at?.(-1)?.total ?? 0) - (expenses?.at?.(-1)?.total ?? 0)

  return (
        <>
            <Header year={year ?? ''}/>
            <Grid container spacing={3} mb={3}>
                <Grid size={{ xs: 12, md: 4 }}>
                    <YearCard title='Ingresos' data={incomes?.at?.(-1)?.total} color='info'/>
                </Grid>
                <Grid size={{ xs: 12, md: 4 }}>
                    <YearCard title='Gastos' data={expenses?.at?.(-1)?.total} color='error'/>
                </Grid>
                <Grid size={{ xs: 12, md: 4 }}>
                    <YearCard title='Balance' data={balance} color={balance < 0 ? 'warning' : 'success'}/>
                </Grid>
            </Grid>
            <Typography variant="h5">GASTOS</Typography>
            <MainCard sx={{ mt: 2 }} content={false}>
                <YearTable data={expenses}/>
            </MainCard>
            <Typography variant="h5" mt={2}>INGRESOS</Typography>
            <MainCard sx={{ mt: 2 }} content={false}>
                <YearTable data={incomes}/>
            </MainCard>

        </>
  )
}

export default Budgets
