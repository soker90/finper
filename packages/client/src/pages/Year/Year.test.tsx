// @vitest-environment happy-dom
import { describe, expect, it } from 'vitest'
import { render } from '../../test/testUtils'
import Year from './index'

describe('Year', () => {
  it('renders without crashing', () => {
    const { container } = render(<Year/>)

    expect(container).toMatchSnapshot()
  })
})
