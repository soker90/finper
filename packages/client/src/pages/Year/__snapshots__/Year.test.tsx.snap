// Vitest Snapshot v1

exports[`Year > renders without crashing 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-4rifo3-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-oqu2pi-MuiButtonBase-root-MuiButton-root"
        href="/anual/2022"
        tabindex="0"
      >
         ACTUAL
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-1l0hcb1-MuiGrid-root"
    >
      <div
        class="css-1p1m4ay"
      >
        <a
          aria-disabled="true"
          aria-label="izquierda"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-172ns19-MuiButtonBase-root-MuiIconButton-root"
          href="/anual/-1"
          tabindex="-1"
        >
          <span
            aria-label="left"
            class="anticon anticon-left"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
        </a>
        <p
          class="MuiTypography-root MuiTypography-body1 css-oub1po-MuiTypography-root"
        >
           
        </p>
        <a
          aria-disabled="true"
          aria-label="derecha"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-172ns19-MuiButtonBase-root-MuiIconButton-root"
          data-testid="right-arrow"
          href="/anual/1"
          tabindex="-1"
        >
          <span
            aria-label="right"
            class="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    />
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18lu0av-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lwnmxi-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-13mu4hp-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-1uvcgxq-MuiCardContent-root"
        >
          <div
            class="css-zgbp5c-MuiStack-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-oym2bt-MuiTypography-root"
            >
              Ingresos
            </h6>
            <div
              class="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4 css-p7gs2c-MuiTypography-root"
                >
                  0 €
                </h4>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lwnmxi-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-13mu4hp-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-1uvcgxq-MuiCardContent-root"
        >
          <div
            class="css-zgbp5c-MuiStack-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-oym2bt-MuiTypography-root"
            >
              Gastos
            </h6>
            <div
              class="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4 css-ysl4mw-MuiTypography-root"
                >
                  0 €
                </h4>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lwnmxi-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-13mu4hp-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-1uvcgxq-MuiCardContent-root"
        >
          <div
            class="css-zgbp5c-MuiStack-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-oym2bt-MuiTypography-root"
            >
              Balance
            </h6>
            <div
              class="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4 css-ah5xzx-MuiTypography-root"
                >
                  0 €
                </h4>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <h5
    class="MuiTypography-root MuiTypography-h5 css-1rit3o0-MuiTypography-root"
  >
    GASTOS
  </h5>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-15oiqrv-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiTableContainer-root css-1hmt0ej-MuiTableContainer-root"
    >
      <table
        aria-labelledby="tableTitle"
        class="MuiTable-root MuiTable-stickyHeader css-2przaq-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-19zgueb-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1j77n0n-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Categoria
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Enero
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Febrero
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Marzo
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Abril
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Mayo
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Junio
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Julio
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Agosto
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Septiembre
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Octubre
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Noviembre
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Diciembre
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Totales
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-rdmsxo-MuiTableBody-root"
        />
      </table>
    </div>
  </div>
  <h5
    class="MuiTypography-root MuiTypography-h5 css-1h1neps-MuiTypography-root"
  >
    INGRESOS
  </h5>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-15oiqrv-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiTableContainer-root css-1hmt0ej-MuiTableContainer-root"
    >
      <table
        aria-labelledby="tableTitle"
        class="MuiTable-root MuiTable-stickyHeader css-2przaq-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-19zgueb-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1j77n0n-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Categoria
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Enero
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Febrero
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Marzo
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Abril
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Mayo
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Junio
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Julio
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Agosto
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Septiembre
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Octubre
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Noviembre
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Diciembre
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-2ziqjn-MuiTableCell-root"
              scope="col"
            >
              Totales
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-rdmsxo-MuiTableBody-root"
        />
      </table>
    </div>
  </div>
</div>
`;
